src/common/DOMWorld.ts

```diff
@@ -568,6 +587,29 @@ export class DOMWorld {
       }
     }
   }
+
+  waitForFunction(
+    pageFunction: Function | string,
+    options: { polling?: string | number; timeout?: number } = {},
+    ...args: SerializableOrJSHandle[]
+  ): Promise<JSHandle> {
+    const {
+      polling = 'raf',
+      timeout = this._timeoutSettings.timeout(),
+    } = options;
+    return new WaitTask(
+      this,
+      pageFunction,
+      'function',
+      polling,
+      timeout,
+      ...args
+    ).promise;
```

Comment:
one day (not in this PR!) I'm going to refactor `WaitTask` to take an options bag, it's so hard to figure out what all of these are for!

Reply:
I agree. At least we get rid of the `predicateQueryHandlerBody` argument with this change 🙂